build.rs

```diff
@@ -460,7 +470,7 @@ fn build_c_code(
 
         // For Windows we also pregenerate the object files for non-Git builds so
         // the user doesn't need to install the assembler.
-        if use_pregenerated && target.os == WINDOWS {
+        if use_pregenerated && (target.os == WINDOWS || target.os == UEFI) {
```

Comment:
I suggest we add a:

```
fn supports_preassembly(arch: &str, os: &str) -> bool {
    ASM_TARGETS.iter().find(|asm_target| {
         asm_target.preassemble && asm_target.arch == arch && asm_target.oss.contains(os)
   }).is_some()
)
```

This will fix a bug in the existing logic wherein we don't screen out aarch64 Windows, for which we just added support, where we don't preassemble.